docs/core/whats-new/dotnet-9/libraries.md

```diff
@@ -184,22 +271,318 @@ The following code snippet shows some example usage.
+    public ICustomAttributeProvider? AttributeProvider { get; }
+}
+```
+
+## Spans
+
+In high-performance code, spans are often used to avoid allocating strings unnecessarily. <xref:System.Span%601> and <xref:System.ReadOnlySpan%601> continue to revolutionize how code is written in .NET, and every release more and more methods are added that operate on spans. .NET 9 includes the following span-related updates:
+
+- [File helpers](#file-helpers)
+- [`params ReadOnlySpan<T>` overloads](#params-readonlyspant-overloads)
+
+### File helpers
+
+The <xref:System.IO.File> class now has new helpers to easily and directly write `ReadOnlySpan<char>`/`ReadOnlySpan<byte>` and `ReadOnlyMemory<char>`/`ReadOnlyMemory<byte>` to files.
+
+The following code efficiently writes a `ReadOnlySpan<char>` to a file.
+
+```csharp
+ReadOnlySpan<char> text = ...;
+File.WriteAllText(filePath, text);
+```
+
+New <xref:System.ReadOnlySpan.StartsWith%2A> and <xref:System.ReadOnlySpan.EndsWith%2A> extension methods have also been added for spans, making it easy to test whether a <xref:System.ReadOnlySpan%601> starts or ends with a specific `T` value.
+
+The following code uses these new convenience APIs.
+
+:::code language="csharp" source="../snippets/dotnet-9/csharp/Spans.cs" id="StartsWith":::
+
+### `params ReadOnlySpan<T>` overloads
+
+C# has always supported marking array parameters as [`params`](../../../csharp/language-reference/keywords/method-parameters.md#params-modifier). This keyword enables a simplified calling syntax. For example, the <xref:System.String.Join(System.String,System.String[])?displayProperty=nameWithType> method's second parameter is marked with `params`. You can call this overload with an array or by passing the values individually:
+
+```csharp
+string result = string.Join(", ", new string[3] { "a", "b", "c" });
+string result = string.Join(", ", "a", "b", "c");
+```
+
+Prior to .NET 9, when you pass the values individually, the C# compiler emits code identical to the first call by producing an implicit array around the three arguments.
+
+Starting in C# 13, you can use `params` with any argument that can be constructed via a collection expression, including spans (<xref:System.Span%601> and <xref:System.ReadOnlySpan%601>). That's beneficial for a variety of reasons, including performance. The C# compiler can store the arguments on the stack, wrap a span around them, and pass that off to the method, which avoids the implicit array allocation that would have otherwise resulted.
+
+.NET 9 now includes over 60 methods with a `params ReadOnlySpan<T>` parameter. Some are brand new overloads, and some are existing methods that already took a `ReadOnlySpan<T>` but now have that parameter marked with `params`. The net effect is if you upgrade to .NET 9 and recompile your code, you'll see performance improvements without making any code changes. That's because the compiler prefers to bind to span-based overloads than to the array-based overloads.
```

Comment:
```suggestion
.NET 9 includes over 60 methods with a `params ReadOnlySpan<T>` parameter. Some are brand new overloads, and some are existing methods that already took a `ReadOnlySpan<T>` but now have that parameter marked with `params`. The net effect is if you upgrade to .NET 9 and recompile your code, you'll see performance improvements without making any code changes. That's because the compiler prefers to bind to span-based overloads than to the array-based overloads.
```